### ISSUE TITLE
Failed to start installation phase bootstrapping

### ISSUE THREAD

**Issue by jackiehjm** [Wind-River/meta-akraino#4]
```
[2020-02-04 13:30:14,914   DEBUG cmbackendhandler( 46)] set_property called for setting cloud.installation_phase="setup-ended"
[2020-02-04 13:30:14,915   DEBUG cmbackendhandler( 46)] set_property called for setting cloud.installation_phase="bootstrapping-started"
^[[0;35mVariable files: "-e @/etc/openstack_deploy/user_secrets.yml -e @/etc/openstack_deploy/user_variables.yml  --skip-tags pip,install-apt,install-yum,memcached_server-install,haproxy_package-install,galera-server-pip-packages,galera-gpg-keys,galera-repos,galera-pre-yum-packages,galera-client-install,galera_server-install,rabbitmq-yum-packages,rabbitmq-plugin-config,common-log,bird,rsyslog-yum-packages,glance-install,cinder-install,nova-install,nova-pip-packages,nova-novnc-git,calico-pip-packages,dragonflow-pip-packages,neutron-pkg-install,heat-pkg-install,horizon-pkg-install,swift-pkg-install,swift-chk-hashes,ironic-l2,ironic-pkg-install,ceph-install,package-install -e @/etc/openstack_deploy/env.d/baremetal.yml -e @/etc/openstack_deploy/user_variables.yml -e @/etc/openstack_deploy/user_secrets.yml"^[[0m
 [WARNING]:  * Failed to parse /opt/openstack-ansible/inventory/inventory.sh
with script plugin: Inventory script (/opt/openstack-
ansible/inventory/inventory.sh) had an execution error: Traceback (most recent
call last):   File "/usr/local/bin/cmcli", line 6, in <module>     from
pkg_resources import load_entry_point   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 3098, in <module>     @_call_aside
File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3082,
in _call_aside     f(*args, **kwargs)   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 3111, in
_initialize_master_working_set     working_set = WorkingSet._build_master()
File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 564, in
_build_master     ws = cls()   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 557, in __init__
self.add_entry(entry)   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 613, in add_entry     for dist in
find_distributions(entry, True):   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 1961, in find_on_path     for dist in
factory(fullpath):   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 2031, in distributions_from_metadata
root, entry, metadata, precedence=DEVELOP_DIST,   File "/usr/lib/python2.7
/site-packages/pkg_resources/__init__.py", line 2470, in from_location
py_version=py_version, platform=platform, **kw   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 2845, in _reload_version
md_version = _version_from_file(self._get_metadata(self.PKG_INFO))   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2434, in
_version_from_file     line = next(iter(version_lines), '')   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2624, in
_get_metadata     for line in self.get_metadata_lines(name):   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1407, in
get_metadata_lines     return yield_lines(self.get_metadata(name))   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1403, in
get_metadata     value = self._get(self._fn(self.egg_info, name))   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1515, in
_get     with open(path, 'rb') as stream: IOError: [Errno 13] Permission
denied: '/usr/lib/python2.7/site-packages/Flask_Bootstrap-3.3.7.1-py2.7.egg-
info/PKG-INFO'
 [WARNING]:  * Failed to parse /opt/openstack-ansible/inventory/inventory.sh
with ini plugin: /opt/openstack-ansible/inventory/inventory.sh:17: Expected
key=value host variable assignment, got: [
 [WARNING]: Unable to parse /opt/openstack-ansible/inventory/inventory.sh as an
inventory source
[DEPRECATION WARNING]: The use of 'include' for tasks has been deprecated. Use
'import_tasks' for static inclusions or 'include_tasks' for dynamic inclusions.
 This feature will be removed in a future release. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: include is kept for backwards compatibility but usage is
 discouraged. The module documentation details page may explain more about this
 rationale.. This feature will be removed in a future release. Deprecation
warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: The use of 'static' has been deprecated. Use
'import_tasks' for static inclusion, or 'include_tasks' for dynamic inclusion.
This feature will be removed in a future release. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
 [WARNING]: Could not match supplied host pattern, ignoring: management

PLAY [Report installation progress] ********************************************
skipping: no hosts matched
 [WARNING]: Could not match supplied host pattern, ignoring: baremetal-
infra_hosts

PLAY [Power-off all the hosts provided in user_config.yml before starting deployemt.] ***
skipping: no hosts matched

PLAY [Report installation progress] ********************************************
skipping: no hosts matched
 [WARNING]: Could not match supplied host pattern, ignoring: shared-infra_hosts

PLAY [Configure NTP service on controllers] ************************************
skipping: no hosts matched
 [WARNING]: Could not match supplied host pattern, ignoring: base
PLAY [Report installation progress] ********************************************
skipping: no hosts matched
 [WARNING]: Could not match supplied host pattern, ignoring: rsyslog

PLAY [Install rsyslog] *********************************************************
skipping: no hosts matched

PLAY [Report installation progress] ********************************************
skipping: no hosts matched
 [WARNING]: Could not match supplied host pattern, ignoring:
baremetal_management_nodes

PLAY [Installation and setup of Ironic] ****************************************
skipping: no hosts matched

PLAY RECAP *********************************************************************

Tuesday 04 February 2020  13:30:16 +0000 (0:00:00.063)       0:00:00.063 ******
===============================================================================
[2020-02-04 13:30:16,156   DEBUG cmbackendhandler( 46)] set_property called for setting cloud.installation_phase="bootstrapping-ended"
[2020-02-04 13:30:16,157   DEBUG cmbackendhandler( 46)] set_property called for setting cloud.installation_phase="provisioning-started"
^[[0;35mVariable files: "-e @/etc/openstack_deploy/user_secrets.yml -e @/etc/openstack_deploy/user_variables.yml  --skip-tags pip,install-apt,install-yum,memcached_server-install,haproxy_package-install,galera-server-pip-packages,galera-gpg-keys,galera-repos,galera-pre-yum-packages,galera-client-install,galera_server-install,rabbitmq-yum-packages,rabbitmq-plugin-config,common-log,bird,rsyslog-yum-packages,glance-install,cinder-install,nova-install,nova-pip-packages,nova-novnc-git,calico-pip-packages,dragonflow-pip-packages,neutron-pkg-install,heat-pkg-install,horizon-pkg-install,swift-pkg-install,swift-chk-hashes,ironic-l2,ironic-pkg-install,ceph-install,package-install -e @/etc/openstack_deploy/env.d/baremetal.yml -e @/etc/openstack_deploy/user_variables.yml -e @/etc/openstack_deploy/user_secrets.yml"^[[0m
 [WARNING]:  * Failed to parse /opt/openstack-ansible/inventory/inventory.sh
with script plugin: Inventory script (/opt/openstack-
ansible/inventory/inventory.sh) had an execution error: Traceback (most recent
call last):   File "/usr/local/bin/cmcli", line 6, in <module>     from
pkg_resources import load_entry_point   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 3098, in <module>     @_call_aside
File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3082,
in _call_aside     f(*args, **kwargs)   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 3111, in
_initialize_master_working_set     working_set = WorkingSet._build_master()
File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 564, in
_build_master     ws = cls()   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 557, in __init__
self.add_entry(entry)   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 613, in add_entry     for dist in
find_distributions(entry, True):   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 1961, in find_on_path     for dist in
factory(fullpath):   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 2031, in distributions_from_metadata
root, entry, metadata, precedence=DEVELOP_DIST,   File "/usr/lib/python2.7
/site-packages/pkg_resources/__init__.py", line 2470, in from_location
py_version=py_version, platform=platform, **kw   File "/usr/lib/python2.7/site-
packages/pkg_resources/__init__.py", line 2845, in _reload_version
md_version = _version_from_file(self._get_metadata(self.PKG_INFO))   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2434, in
_version_from_file     line = next(iter(version_lines), '')   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2624, in
_get_metadata     for line in self.get_metadata_lines(name):   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1407, in
get_metadata_lines     return yield_lines(self.get_metadata(name))   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1403, in
get_metadata     value = self._get(self._fn(self.egg_info, name))   File
"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1515, in
_get     with open(path, 'rb') as stream: IOError: [Errno 13] Permission
denied: '/usr/lib/python2.7/site-packages/Flask_Bootstrap-3.3.7.1-py2.7.egg-
info/PKG-INFO'
 [WARNING]:  * Failed to parse /opt/openstack-ansible/inventory/inventory.sh
with ini plugin: /opt/openstack-ansible/inventory/inventory.sh:17: Expected
key=value host variable assignment, got: [
 [WARNING]: Unable to parse /opt/openstack-ansible/inventory/inventory.sh as an
inventory source
[DEPRECATION WARNING]: The use of 'include' for tasks has been deprecated. Use
'import_tasks' for static inclusions or 'include_tasks' for dynamic inclusions.
 This feature will be removed in a future release. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: include is kept for backwards compatibility but usage is
 discouraged. The module documentation details page may explain more about this
 rationale.. This feature will be removed in a future release. Deprecation
warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: 'include' for playbook includes. You should use
'import_playbook' instead. This feature will be removed in version 2.8.
Deprecation warnings can be disabled by setting deprecation_warnings=False in
ansible.cfg.
ERROR! the role 'ceph-defaults' was not found in /opt/openstack-ansible/playbooks/roles:/etc/ansible/roles:/opt/openstack-ansible/playbooks/roles:/opt/openstack-ansible/playbooks/roles:/opt/openstack-ansible/playbooks
The error appears to have been in '/opt/openstack-ansible/playbooks/site.yml': line 74, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  roles:
    - ceph-defaults
      ^ here
[2020-02-04 13:30:17,282   ERROR cmactivateserverhandler(100)] Skipping installationactivator, got exception Playbook provisioning-playbook.yml failed

```

**Comment by jackiehjm** [comment]
After #15 is fixed:

```
ERROR! the role 'post_install_caas_infra' was not found in /opt/openstack-ansible/playbooks/roles:/etc/ansible/roles:/opt/openstack-ansible/playbooks/roles:/opt/openstack-ansible/playbooks/roles:/opt/openstack-ansible/playbooks

The error appears to have been in '/opt/openstack-ansible/playbooks/install_caas_infra.yaml': line 36, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  roles:
    - post_install_caas_infra
      ^ here
[2020-02-06 11:53:54,980   ERROR cmactivateserverhandler(100)] Skipping installationactivator, got exception Playbook postconfig-playbook.yml failed

```

**Comment by jackiehjm** [comment]
```
Permission denied: '/usr/lib/python2.7/site-packages/Flask_Bootstrap-3.3.7.1-py2.7.egg-info/PKG-INFO'
Permission denied: '/usr/lib/python2.7/site-packages/docker_registry_core-2.0.3-py2.7.egg-info/PKG-INFO'
```

**Comment by jackiehjm** [comment]
> 
> 
> After #15 is fixed:
> 
> ```
> ERROR! the role 'post_install_caas_infra' was not found in /opt/openstack-ansible/playbooks/roles:/etc/ansible/roles:/opt/openstack-ansible/playbooks/roles:/opt/openstack-ansible/playbooks/roles:/opt/openstack-ansible/playbooks
> 
> The error appears to have been in '/opt/openstack-ansible/playbooks/install_caas_infra.yaml': line 36, column 7, but may
> be elsewhere in the file depending on the exact syntax problem.
> 
> The offending line appears to be:
> 
>   roles:
>     - post_install_caas_infra
>       ^ here
> [2020-02-06 11:53:54,980   ERROR cmactivateserverhandler(100)] Skipping installationactivator, got exception Playbook postconfig-playbook.yml failed
> ```

In REC build185:
```
[newadmin@controller-1 ~]$ rpm -qf /etc/ansible/roles/post_install_caas_infra
caas-infra-charts-1.0.0-34.el7.centos.ta.noarch
```

Open new issue for this: #33

**Comment by jackiehjm** [comment]
> 
> 
> ```
> Permission denied: '/usr/lib/python2.7/site-packages/Flask_Bootstrap-3.3.7.1-py2.7.egg-info/PKG-INFO'
> Permission denied: '/usr/lib/python2.7/site-packages/docker_registry_core-2.0.3-py2.7.egg-info/PKG-INFO'
> ```

Open a new issue for this: #32

**Comment by jackiehjm** [comment]
Fixed by sevaral other issue: #15, #32, #33, #37, #38, #36
